torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -1644,6 +1655,28 @@ def _linear_fp8_act_fp8_weight_impl(
         use_fast_accum=scaled_mm_config.use_fast_accum,
     ).reshape(out_shape)
 
+def _linear_fp_act_fp8_weight_check(
```

Comment:
this is separate I guess, would be good to put this in a different PR?

Reply:
This is needed to run float8 weight-only kernel, and support AQT tensor parallel.

Reply:
I thought we already have fallback? https://github.com/pytorch/ao/blob/0ffbf856a2b61ed4dfb2c28376e8f4e801c9b962/torchao/dtypes/affine_quantized_tensor.py#L1723-L1730

Reply:
https://github.com/pytorch/ao/blob/0ffbf856a2b61ed4dfb2c28376e8f4e801c9b962/torchao/dtypes/affine_quantized_tensor.py#L1691

It check through the dispatchers, to find a kernel, and if it doesn't find one, it raise a no dispatch kernel found error. Hence need a kernel in the dispatch table.

Reply:
not sure why this is required, but we can debug separately I think